### PR TITLE
Inductor cpp wrapper: support MKLPackedLinear

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89743
* #94977
* #94964
* #97655
* __->__ #90755

Invoke `torch.ops.mkl._mkl_linear` from c++.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]
LGTM except for some small nits.

File: torch/_inductor/codegen/wrapper.py [hunk 1]
```diff
@@ -892,6 +905,33 @@ def make_buffer_allocation(self, buffer):
...
 
+    def generate_fusion_ops_code(
+        self,
+        name,
+        kernel,
+        cpp_kernel,
+        codegen_args,
+        cpp_op_schema,
+        cpp_kernel_key,
+        cpp_kernel_overlad_name="",
```
[the remark this answers is not in the document]
Comment (chunyuan-w) [reply torch/_inductor/codegen/wrapper.py]: Fixed the typo

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -3357,14 +3362,36 @@ def __init__(
...
+        else:
+            args = self.codegen_args()
+
+        wrapper.generate_fusion_ops_code(
+            self.get_name(),
+            self.kernel,
+            self.cpp_kernel,
+            args,
+            self.cpp_op_schema,
+            self.cpp_kernel_key,
```
[the remark this answers is not in the document]
Comment (chunyuan-w) [reply torch/_inductor/ir.py:3394]: Yes, all the fusion OPs share this logic. I'll submit a following up PR to refactor this part out into a common place.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**